vcd/resource_vcd_vapp_network.go

```diff
@@ -515,7 +515,7 @@ func resourceVcdVappNetworkImport(d *schema.ResourceData, meta interface{}) ([]*
 	}
 
 	if vappNetworkToImport.Configuration.FenceMode == types.FenceModeBridged ||
-		(vappNetworkToImport.Configuration.FenceMode == types.FenceModeNAT && vappNetworkToImport.Configuration.Features.DhcpService == nil) {
+		(vappNetworkToImport.Configuration.FenceMode == types.FenceModeNAT && vappNetworkToImport.Configuration.IPScopes.IPScope[0].IsInherited == true) {
```

Comment:
Please add a comment why we check for `vappNetworkToImport.Configuration.IPScopes.IPScope[0].IsInherited == true` - feels like we're trying to figure out something more than what it looks like from code.

Same ask for identical lines in other files.

Actually, can this check become a function with a descriptive name and shared across all those files?

Reply:
As @lvirbalas mentioned - probably this could become a function.

Also - you are a lot deeper into this now, but wondering - wouldn't `NetworkConfiguration.ParentNetwork != nil` be a "closer" hint?

Reply:
done